examples/threading/threading.c

```diff
@@ -28,6 +69,30 @@ bool start_thread_obtaining_mutex(pthread_t *thread, pthread_mutex_t *mutex,int
      * 
      * See implementation details in threading.h file comment block
      */
-    return false;
+     
+    //allocate memory for thread parameter
+	struct thread_data* thread_data_params = (struct thread_data*) malloc(sizeof(struct thread_data));
+	//check for memory allocation error
+	if(thread_data_params == NULL){
+		ERROR_LOG("Malloc failed");
+		return false;
+	}
+	
+	//assign struct parameters
+	thread_data_params->wait_to_obtain_ms = wait_to_obtain_ms;
+	thread_data_params->wait_to_release_ms = wait_to_release_ms;
+	thread_data_params->mutex_thread = mutex;
+	
+	//create thread
+	int rc = pthread_create(thread, NULL, threadfunc,(void *) thread_data_params);
+	//check for error in creation and return true if success
+	if(rc != 0){
+		ERROR_LOG("Thread creation failed");
+		return false;
```

Comment:
@vishalraj3112 who frees `thread_data_params` in this code path?

Reply:
The thread_data_params is freeded by the caller itself right? Thats why didnt free it.